src/transformers/models/idefics/processing_idefics.py

```diff
@@ -311,18 +318,62 @@ def __call__(
                 transforms.Normalize(mean=self.image_mean, std=self.image_std),
             ]
         )
-        inputs = processor(prompts, transform=image_transform, return_tensors="pt")
+        inputs = processor(text=prompts, transform=image_transform, return_tensors="pt")
         ```
 
         In order to help debug prompt generation enable `debug=True` which will show you what's happening.
 
         """
+        if images is None and text is None:
+            raise ValueError("You need to specify either `text` or `images` and `text`.")
+        # for BC
+        if text is None:
+            # if the user didn't specify text=text in the call, we assume they want to use the old behavior
+            # with text (previously prompts) as a first argument
+            warnings.warn(
+                "The use of `text` as the first argument will be deprecated in the future. `images` is now the first argument."
+                "The first given argument will be considered as `prompts` in the old behavior.",
+            )
+            text = images
+            images = None
+        if images is None:
+            # assuming the user wants to use the old behavior with prompts as the only argument
+            prompts = text
+        elif text is not None:
+            # Assuming image-text-to-text behavior:
+            # Check if batched images are provided
+            if not isinstance(images, (list, tuple)):
+                images = [images]
+            if isinstance(text, str):
+                # one prompt for all images instead of one prompt per image
+                text = [text] * len(images)
+            # Check if batched text is provided
+            if isinstance(text, (list, tuple)) and len(text) != len(images):
+                raise ValueError(
+                    "When using the image-text-to-text behavior, the number of prompts should be the same as the number of images."
+                )
+            # Check that only text is present in the prompts
+            if not all(isinstance(i, str) for i in text):
+                raise ValueError("When using the image-text-to-text behavior, the prompts should only contain text.")
+            prompts = list(zip(images, text))
```

Comment:
Lots of logic is needed for backward compatibility, as idefics used to take only `prompts` where text and images inputs would be interleaved. This added logic preserve supports for these kind of inputs (where `prompts` is replaced by `text` arg), while adding support for usual text and images inputs as in other image-text-to-text models. This will also be useful to support idefics in the image-text-to-text pipeline.